packages/shared/src/components/modals/bookmark/BookmarkFolderModal.tsx

```diff
@@ -0,0 +1,221 @@
+  onSubmit: (name: string) => void;
+  // TODO: Replace with correct type when implementing request
+  folder?: {
+    name: string;
+    id: string;
+    icon?: string;
+  };
+  folderCount?: number;
+};
+
+const emojiOptions = [
+  '',
+  '🐹',
+  '🐍',
+  '☕️',
+  '🔥',
+  '📦',
+  '⚙️',
+  '🐙',
+  '🐳',
+  '💡',
+  '📜',
+  '🚀',
+];
+
+const createFirstFolder =
+  'You can create your first folder for free! Organize your bookmarks and see how it works. To unlock unlimited folders,';
+const getMoreFolders =
+  "You've used your free folder. To create more and keep your bookmarks perfectly organized,";
+
+const PlusCTA = ({ folderCount }: { folderCount: number }) => {
+  const { logSubscriptionEvent } = usePlusSubscription();
+
+  return (
+    <Typography type={TypographyType.Callout} color={TypographyColor.Secondary}>
+      <span>{folderCount === 0 ? createFirstFolder : getMoreFolders}</span>{' '}
+      <Link
+        target="_blank"
+        href={plusUrl}
+        rel={anchorDefaultRel}
+        onClick={() => {
+          logSubscriptionEvent({
+            event_name: LogEvent.UpgradeSubscription,
+            target_id: TargetId.BookmarkFolder,
+          });
+        }}
+      >
+        <Typography
+          tag={TypographyTag.Span}
+          type={TypographyType.Callout}
+          color={TypographyColor.Plus}
+          className="underline"
+        >
+          upgrade to Plus
+        </Typography>
+      </Link>
+    </Typography>
+  );
+};
+
+const FormActions = ({
+  valid,
+  folderCount,
+  isPlus,
+  isMobile,
+}: {
+  valid: boolean;
+  folderCount: number;
+  isPlus: boolean;
+  isMobile: boolean;
+}) => {
+  const { closeModal } = useLazyModal();
+
+  return (
+    <div
+      className={classNames(
+        'flex w-full justify-between px-4 pt-4 tablet:px-0 tablet:pt-0',
+        isMobile && 'border-b border-border-subtlest-tertiary pb-4',
+      )}
+    >
+      {isMobile && (
+        <Button
+          type="button"
+          onClick={closeModal}
+          variant={ButtonVariant.Tertiary}
+        >
+          Cancel
+        </Button>
+      )}
+      <Button
+        type="submit"
+        className="tablet:flex-grow"
+        disabled={!valid}
+        variant={ButtonVariant.Primary}
+      >
+        {!isPlus && folderCount > 0 ? (
+          <>
+            <DevPlusIcon /> Upgrade to plus
+          </>
+        ) : (
+          'Create folder'
+        )}
+      </Button>
+    </div>
+  );
+};
+
+const BookmarkFolderModal = ({
+  folder,
+  onSubmit,
+  folderCount = 0,
+  ...rest
+}: BookmarkFolderModalProps): ReactElement => {
+  const { isPlus } = usePlusSubscription();
+  const [valid, setValid] = useState(false);
+  const isMobile = useViewSize(ViewSize.MobileL);
+
+  const handleSubmit = (e: FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    // TODO: Dummy component for now. Implement request later.
+    // const formJson = formToJson(e.currentTarget);
+  };
```

Comment:
We should fire `onSubmit({ ...folder, icon, name })` here.

Reply:
Note the TODO :)

Reply:
We should add this method call here, this component is dummy but still have to call the prop function if passed any.
 
We make it dummy because we can use both for create/edit on need, but we definitely should handle the function call